base/compiler/abstractinterpretation.jl

```diff
@@ -853,14 +847,8 @@ function concrete_eval_eligible(interp::AbstractInterpreter,
             return :none
         end
     end
-    if !effects.noinbounds && stmt_taints_inbounds_consistency(sv)
-        # If the current statement is @inbounds or we propagate inbounds,
-        # the call's :consistent-cy is tainted and not consteval eligible.
-        add_remark!(interp, sv, "[constprop] Concrete evel disabled for inbounds")
-        return :none
-    end
     mi = result.edge
-    if mi !== nothing && is_foldable(effects)
+    if mi !== nothing && is_foldable(effects, !stmt_taints_inbounds_consistency(sv))
```

Comment:
Maybe we need a similar change to `concrete_eval_invoke`?

Reply:
Actually, I don't think it's required here at all. Both concrete eval and semi concrete eval are safe here independent of the state of the inbounds flag.